lib/OpenQA/WebAPI/Controller/Test.pm

```diff
@@ -142,28 +141,25 @@ sub list_running_ajax {
         ],
     );
 
-    my @running;
-    while (my $job = $running->next) {
-        my $job_id = $job->id;
-        push(
-            @running,
-            {
-                DT_RowId => 'job_' . $job_id,
-                id => $job_id,
-                clone => $job->clone_id,
-                test => $job->TEST . '@' . ($job->MACHINE // ''),
-                distri => $job->DISTRI // '',
-                version => $job->VERSION // '',
-                flavor => $job->FLAVOR // '',
-                arch => $job->ARCH // '',
-                build => $job->BUILD // '',
-                testtime => ($job->t_started // '') . 'Z',
-                result => $job->result,
-                group => $job->group_id,
-                state => $job->state,
-                progress => $job->progress_info,
-            });
-    }
+    my @running = map {
```

Comment:
Hm, this feels like a dejavu. I don't see the benefit of the change. Haven't you proposed this change before? Maybe it was similar code elsewhere. 
Have you benchmarked this and looked at the memory consumption? The new code will load all database objects into memory.

Reply:
Yes, could very well be that I have proposed this change in before. Sorry about that. I have excluded it for now from the commit. Still, I think we should be able to process all objects in memory. I consider this cleaner with such functional rather than procedural code but I have not benchmarked it.